packages/context/src/binding-inspector.ts

```diff
@@ -215,12 +220,23 @@ export function createBindingFromClass<T = unknown>(
   cls: Constructor<T | Provider<T>>,
   options: BindingFromClassOptions = {},
 ): Binding<T> {
-  const templateFn = bindingTemplateFor(cls);
-  let key = options.key;
-  if (!key) {
-    key = buildBindingKey(cls, options);
```

Comment:
There is no need to check `options.key` here, it's already handled by `buildBindingKey`.